src/System.Private.CoreLib/shared/System/Text/StringBuilder.cs

```diff
@@ -1203,7 +1203,7 @@ public StringBuilder Append(char[]? value)
             {
                 unsafe
                 {
-                    fixed (char* valueChars = &value[0])
+                    fixed (char* valueChars = &value![0])
```

Comment:
This to me looks like a compiler bug in the data flow analysis... the `if (value?.Length > 0)` condition shouldn't be true if `value is null` because it is guarded using `?` -- cc: @jaredpar @gafter

Reply:
Also, I'm curious, why did you have to add this if it is building fine for everybody else's in their computer and on CI.

Reply:
The latest build of VS gets this right and the `!` should not be needed.

---
In reply to: [273621720](https://github.com/dotnet/coreclr/pull/23489#discussion_r273621720) [](ancestors = 273621720)